app/time_tracking/work_statistic/views.py

```diff
@@ -0,0 +1,148 @@
+def quarter_statistics_detail(request, format=None):
+    """
+    Retrieve work time since a 13 weeks ago.
+    """
+    if request.method == 'GET':
+        work_times = WorkTime.objects.filter(
+            owner=request.user,
+            work_time__isnull=False,
+            end_time__gt=timezone.now() - timedelta(weeks=13)
+        ).aggregate(Sum('work_time'))
+        responce = {
+            "total_working_time_in_seconds": work_times['work_time__sum']
+        }
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def year_statistics_detail(request, format=None):
+    """
+    Retrieve work time since a 356 days ago.
+    """
+    if request.method == 'GET':
+        work_times = WorkTime.objects.filter(
+            owner=request.user,
+            work_time__isnull=False,
+            end_time__gt=timezone.now() - timedelta(days=356)
+        ).aggregate(Sum('work_time'))
+        responce = {
+            "total_working_time_in_seconds": work_times['work_time__sum']
+        }
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def employees_arrival_and_leving_time_statistics_detail(request, format=None):
+    """
+    Retrieve the employees arrival time and leaving times.
+    """
+    if request.method == 'GET':
+        responce = []
+        for user in User.objects.filter(is_staff=False):
+            avg_work_time_starts = WorkTime.objects.filter(owner=user).aggregate(Avg('start_unix_time'))['start_unix_time__avg']
+            avg_work_time_ends = WorkTime.objects.filter(owner=user).aggregate(Avg('end_unix_time'))['end_unix_time__avg']
+
+            if avg_work_time_starts is not None:
+                avg_work_time_starts = datetime.fromtimestamp(avg_work_time_starts)
+            if avg_work_time_ends is not None:
+                avg_work_time_ends = datetime.fromtimestamp(avg_work_time_ends)
+
+            responce.append({
+                'user': user.username,
+                'avarage_arraival_time': avg_work_time_starts,
+                'avarage_leaving_time': avg_work_time_ends,
+            })
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def working_ours_to_leaving_hours_detail(request, format=None):
+    """
+    Retrieve the team working hours to leaving hours
+    """
+    if request.method == 'GET':
+        working_hours = 0
+        leaving_hours = 0
+        for user in User.objects.filter():
+            work_times_grouped_by_owner_and_date = WorkTime.objects.filter(
+                owner=user,
+                end_unix_time__isnull=False,
+                start_unix_time__isnull=False
```

Comment:
Shouldn't these be prevented from being null in the model level?

Reply:
The `start_unix_time` is not nullable on the model and I'll remove this redundant statement.
but `end_unix_time` is null when the timer isn't finished yet.

Reply:
The useless `start_unix_time__isnull=False` is now removed